### PR TITLE
Conditional variable was replaced with channel notifier

### PR DESCRIPTION
**Pull request by tigrulya-exe**
If we use conditional variable for notifying collector the following situation might happen:

0. Collector has 2 prefix sources: **Source A** and **Source B**.
1. **Source A** notifies about new incoming prefixes.
2. Collector calls `source.Prefixes()` of 2 sources in cycle.
3. **Source B** notifies about new incoming prefixes.
4. Collector writes **Source A**'s prefixes into ConfigMap.
5. Collector blocks on `cond.Wait()`
6. If there is no notifications, collector will never write **Source B**'s prefixes into ConfigMap.

### CODE REVIEW

File: internal/utils/channel_notifier.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	Notify()
+	Wait(ctx context.Context) error
+}
+
+type channelNotifier struct {
+	ch chan struct{}
+}
+
+// NewChannelNotifiable - creates new channelNotifier
+func NewChannelNotifiable() Notifiable {
```
Comment (denis-tingajkin) [inline internal/utils/channel_notifier.go:32]: Let's just use the channel indirect

**Review by denis-tingajkin — CHANGES_REQUESTED** [review]

File: internal/utils/channel_notifier.go (hunk 1 above)
Comment (tigrulya-exe) [reply internal/utils/channel_notifier.go]: Replaced with channel

**Review by denis-tingajkin — APPROVED** [review]
Looks good to me 
@tigrulya-exe Do you plan to cover bug by the unit test to prevent regressions?

**Comment by tigrulya-exe** [comment]
> @tigrulya-exe Do you plan to cover bug by the unit test to prevent regressions?
I think it'll be hard to reproduce such behaviour in unit test. But we have integration test with all sources, when the following situation might happen.

**Comment by denis-tingajkin** [comment]
@tigrulya-exe Can we add some light test prefix sources? I think we should try to cover this issue by unit testing to minimize runs of integration tests.

**Comment by tigrulya-exe** [comment]
@denis-tingajkin added unit test with all sources

**Pull request merged**